File: backend/app/Schemas/resume.py


from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeBase(BaseModel):
    title: str
    file_name: str
    file_path: str
    extracted_text: str | None = None
    skills: str | None = None
    experience: str | None = None
    education: str | None = None
    user_id: int


class ResumeCreate(ResumeBase):
    pass


class ResumeUpdate(BaseModel):
    title: str | None = None
    file_name: str | None = None
    file_path: str | None = None
    extracted_text: str | None = None
    skills: str | None = None
    experience: str | None = None
    education: str | None = None


class ResumeResponse(ResumeBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: careerpilot-frontend/types/resume.ts
export interface Resume {
  id: number;
  title: string;
  file_name: string;
  file_path: string;
  extracted_text: string | null;
  skills: string | null;
  experience: string | null;
  education: string | null;
  user_id: number;
  created_at: string;
}

export interface ResumeUploadResponse {
  message: string;
  resume_id: number;
  filename: string;
  analysis: ATSAnalysis;
}

export interface ATSAnalysis {
  full_name?: string;
  email?: string;
  phone?: string;
  summary?: string;
  skills?: string[];
  experience?: unknown[];
  education?: unknown[];
  projects?: unknown[];
  certifications?: unknown[];
  languages?: unknown[];
  ats_score?: number;
  strengths?: string[];
  weaknesses?: string[];
  missing_keywords?: string[];
  suggestions?: string[];
}
